src/SlackConnector/Connections/Sockets/WebSocketClient.cs

```diff
@@ -32,18 +33,20 @@ public WebSocketClient(IMessageInterpreter interpreter, string url, ProxySetting
 
         public event EventHandler<InboundMessage> OnMessage;
         public event EventHandler OnClose;
-        
-        public async Task Connect()
+
+        public Task Connect()
         {
             var taskSource = new TaskCompletionSource<bool>();
+            EventHandler<ErrorEventArgs> onError = (sender, args) => { taskSource.TrySetException(args.Exception); };
 
-            _webSocket.OnOpen += (sender, args) => { taskSource.SetResult(true); };
-            _webSocket.OnError += (sender, args) => { taskSource.SetException(args.Exception); };
+            _webSocket.OnOpen += (sender, args) => { taskSource.SetResult(true); _webSocket.OnError -= onError; };
```

Comment:
Background:

Initially, what made me do the Ping impl. was that I could never detect if the bot was disconnected. Forgetting about the catch-all try...catches I found that the code in the constructor **throws an exception**.

Why?

The constructor had a TaskCompletionSource there, that was already Completed (since the connection was successful), but the OnError was still active, and at the first error it tried to set the TSC into a Faulted state -- _even though it was already completed_.

So, I did two things, but any of them would suffice.

1. I unsubscribe from the event after the connection attempt is successful. If not, then there is not need to clean up :)
2. Instead of SetXXX I like to use TrySetXXX on the TSC, as usually it's no matter of the Task is _actually_ transitioned into that state. If a previous attempt already transitioned into a different state, well... they won. It's safer for me, as there are no random exceptions, but true, this can hide bugs. 

Here, however, it's safer, as we are working with multi-threaded code of which we have no control over.

What if, 

1. the connection is succesful
2. but then we get disconnected, before we have a chance to unsubscribe from the event? With SetXXX: the app will die from some random unhandled (cannot be handled!) exception. With TrySet: the app will die (usually) when trying to Say something.